examples/threathunter/threat_intelligence/stix_taxii.py

```diff
@@ -0,0 +1,283 @@
+            if want_all or collection.name.lower() in desired_collections:
+                yield from self.import_collection(collection)
+
+    def generate_reports(self):   # NOTE:ignoring binary for now
+        reports = []
+
+        self.create_taxii_client()
+        if not self.client:
+            logging.error('Unable to create taxii client.  Exiting...')
+            return reports
+
+        available_collections = self.query_collections()
+        if not available_collections:
+            logging.warning('Unable to find any collections.  Exiting...')
+            return reports
+
+        reports = self.import_collections(available_collections)
+        if not reports:
+            logging.warning('Unable to import collections.  Exiting...')
+            return reports
+
+        return reports
+
+
+
+
+class StixTaxii():
+    def __init__(self, site_confs):
+        self.config = site_confs
+        self.client = None
+
+    def result(self, **kwargs):
+        """
+        Returns a new AnalysisResult with the given fields populated, updating
+        the database in the background.
+
+        This should be used within the :meth:`analyze` method to create
+        analysis results.
+
+        :rtype: :class:`AnalysisResult`
+
+        Example::
+
+        >>> self.result(analysis_name="foo", score=10)
+        """
+
+        result = AnalysisResult(**kwargs).normalize()
+        return result
+
+
+    def configure_sites(self):
+        self.sites = {}
+        try:
+            for site_name, site_conf in self.config['sites'].items():
+                self.sites[site_name] = TaxiiSiteConnector(site_conf)
+                logging.info(f"loaded site {site_name}")
+        except handled_exceptions as e:
+
+            logging.error(f"Error in parsing config file: {e}")
+
+    def format_report(self, reports):
+        for report in reports:
+            try:
+                analysis_name = report['id']
+                title = report['title']
+                description = report['description']
+                scan_time = datetime.fromtimestamp(report['timestamp'])
+                score = report['score']
+                link = report['link']
+                ioc_dict = report['iocs_v2']
+                result = self.result(
+                                     analysis_name=analysis_name,
+                                     scan_time=scan_time,
+                                     score=score,
+                                     title=title,
+                                     description=description)
+                for ioc_key, ioc_val in ioc_dict.items():
+                    result.attach_ioc_v2(values=ioc_val, field=ioc_key, link=link)
+            except handled_exceptions as e:
+                logging.warning(f"Problem in report formatting: {e}")
+                result = self.result(
+                    analysis_name="exception_format_report", error=True)
+            yield result
+
+    def analyze(self):
```

Comment:
Are we analyzing or are we fetching? I know some of this was pulled from binary analysis so that might be where some of the names came from.

Reply:
No analysis is happening, it's just fetching. A name change would make sense here. Now is called get_send_reports, but open to other suggestions